pkg/client/oci/oci.go

```diff
@@ -0,0 +1,302 @@
+package oci
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"time"
+
+	"github.com/juju/errors"
+	"k8s.io/klog"
+
+	"github.com/bitnami-labs/charts-syncer/api"
+	"github.com/bitnami-labs/charts-syncer/pkg/client/types"
+	"github.com/bitnami-labs/charts-syncer/pkg/helmcli"
+	"github.com/bitnami-labs/charts-syncer/pkg/utils"
+)
+
+const (
+	getTimeout = 1 * time.Minute
+	// HelmChartConfigMediaType is the reserved media type for the Helm chart manifest config
+	HelmChartConfigMediaType = "application/vnd.cncf.helm.config.v1+json"
+
+	// HelmChartContentLayerMediaType is the reserved media type for Helm chart package content
+	HelmChartContentLayerMediaType = "application/tar+gzip"
+)
+
+// Repo allows to operate a chart repository.
+type Repo struct {
+	url      *url.URL
+	username string
+	password string
+}
+
+// Tags contains the tags for a specific OCI artifact
+type Tags struct {
+	Name string
+	Tags []string
+}
+
+// LayerInfo contains info about a OCI layer
+type LayerInfo struct {
+	MediaType string
+	Digest    string
+	Size      int
+}
+
+// TagManifest contains info about the published TAG in OCI
+type TagManifest struct {
+	SchemaVersion int
+	Config        LayerInfo
+	Layers        []LayerInfo
+}
+
+// KnownMediaTypes returns a list of layer mediaTypes that the Helm client knows about
+func KnownMediaTypes() []string {
+	return []string{
+		HelmChartConfigMediaType,
+		HelmChartContentLayerMediaType,
+	}
+}
```

Comment:
check [`ManifestMediaTypes`](https://godoc.org/github.com/docker/distribution#ManifestMediaTypes)

Reply:
actually, I don't need this right now. It's a leftover from a previous test using [oras library](https://github.com/deislabs/oras) to pull/push from/to OCI registries. I think we should be using that library, but I was getting out of time and wanted to have something working first.